src/include/stir/KOSMAPOSL/KOSMAPOSLReconstruction.h

```diff
@@ -150,29 +150,32 @@ class KOSMAPOSLReconstruction:
     {return *this;}
 
   //kernel
-  const std::string get_anatomical_image_filenames() const;
+  const std::vector<std::string> get_anatomical_image_filenames() const;
   const int get_num_neighbours() const;
   const int get_num_non_zero_feat() const;
-  const double get_sigma_m() const;
+  const std::vector<double> get_sigma_m() const;
   const double get_sigma_p() const;
   const double get_sigma_dp() const;
   const double get_sigma_dm()const;
   const bool get_only_2D() const;
   const bool get_hybrid()const;
 
-  shared_ptr<const TargetT> get_anatomical_prior_sptr() const;
+  std::vector<shared_ptr<TargetT> > get_anatomical_prior_sptr();
 
     /*! \name Functions to set parameters
     This can be used as alternative to the parsing mechanism.
    \warning Be careful with setting shared pointers. If you modify the objects in 
    one place, all objects that use the shared pointer will be affected.
   */
 
-  void set_anatomical_prior_sptr(shared_ptr<TargetT>);
+  void set_anatomical_prior_sptr(shared_ptr<TargetT>&, int index);
+  void set_anatomical_image_filenames(const std::string, const int index);
+  void set_anatomical_image_filenames(const std::string);
```

Comment:
I'd prefer these to be `const std::string&`. Porbably makes very little difference, but that's what we use elsewhere in STIR. (should expand our list of coding conventions!)